mkdocs/nav.py

```diff
@@ -14,18 +14,33 @@
 log = logging.getLogger(__name__)
 
 
-def filename_to_title(filename):
+def file_to_title(filename):
     """
     Automatically generate a default title, given a filename.
     """
     if utils.is_homepage(filename):
         return 'Home'
 
+    table_of_contents = None
+    try:
+        with open(filename, 'r') as f:
+            lines = f.read()
+            html_content, table_of_contents, meta = utils.convert_markdown(lines, ['toc'])
+            if len(table_of_contents.items) > 0:
```

Comment:
I've no real opinion on this but since we're already parsing this metadata, I can get that title from metadata. Since it's a supported way of setting the title in markdown it seems only fair that it'd work in this parser?